pkg/controller/kogitotrusty/kogitotrusty_controller_test.go

```diff
@@ -0,0 +1,211 @@
+	}
+	kafkaList := &kafkabetav1.KafkaList{
+		Items: []kafkabetav1.Kafka{
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: "kafka", Namespace: ns},
+				Spec:       kafkabetav1.KafkaSpec{Kafka: kafkabetav1.KafkaClusterSpec{Replicas: 1}},
+				Status: kafkabetav1.KafkaStatus{
+					Listeners: []kafkabetav1.ListenerStatus{
+						{
+							Type:      "plain",
+							Addresses: []kafkabetav1.ListenerAddress{{Host: "kafka", Port: 9092}},
+						},
+					},
+				},
+			},
+		},
+	}
+	cli := test.CreateFakeClient([]runtime.Object{instance, kafkaList}, nil, nil)
+	r := &ReconcileKogitoTrusty{
+		client: cli,
+		scheme: meta.GetRegisteredSchema(),
+	}
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      instance.Name,
+			Namespace: instance.Namespace,
+		},
+	}
+
+	// basic checks
+	res, err := r.Reconcile(req)
+	if err != nil {
+		t.Fatalf("reconcile: (%v)", err)
+	}
+	if !res.Requeue {
+		t.Error("reconcile did not requeue request as expected")
+	}
+
+	// check infra
+	infra, ready, err := infrastructure.EnsureKogitoInfra(ns, cli).WithInfinispan().Apply()
+	assert.NoError(t, err)
+	assert.False(t, ready)  // we don't have status defined since the KogitoInfra controller is not running
+	assert.NotNil(t, infra) // we have a infra instance created during reconciliation phase
+	assert.Equal(t, infrastructure.DefaultKogitoInfraName, infra.GetName())
+	assert.True(t, infra.Spec.InstallInfinispan)
+}
+
+func TestReconcileKogitoTrusty_UpdateHTTPPort(t *testing.T) {
+	ns := t.Name()
+	instance := &v1alpha1.KogitoTrusty{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-trusty",
+			Namespace: ns,
+		},
+		Spec: v1alpha1.KogitoTrustySpec{
+			KogitoServiceSpec: v1alpha1.KogitoServiceSpec{
+				HTTPPort: 9090,
+			},
+			KafkaMeta: v1alpha1.KafkaMeta{
+				KafkaProperties: v1alpha1.KafkaConnectionProperties{
+					UseKogitoInfra: false,
+					ExternalURI:    "my-uri:9022",
+				},
+			},
+			InfinispanMeta: v1alpha1.InfinispanMeta{
+				InfinispanProperties: v1alpha1.InfinispanConnectionProperties{
+					UseKogitoInfra: false,
+					URI:            "another-uri:11222",
+				},
+			},
+		},
+	}
+	is, tag := test.GetImageStreams(infrastructure.DefaultTrustyImageName, instance.Namespace, instance.Name, infrastructure.GetKogitoImageVersion())
+	cli := test.CreateFakeClientOnOpenShift([]runtime.Object{instance, is}, []runtime.Object{tag}, nil)
+	r := &ReconcileKogitoTrusty{
+		client: cli,
+		scheme: meta.GetRegisteredSchema(),
+	}
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      instance.Name,
+			Namespace: instance.Namespace,
+		},
+	}
+
+	// first reconcile
+	_, err := r.Reconcile(req)
+	assert.NoError(t, err)
+
+	// make sure HTTPPort env was added on the deployment
+	deployment := &appsv1.Deployment{}
+	exists, err := kubernetes.ResourceC(cli).FetchWithKey(types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, deployment)
+	assert.True(t, exists)
+	assert.NoError(t, err)
```

Comment:
See: https://github.com/kiegroup/kogito-cloud-operator/blob/master/pkg/test/assert_fetch.go#L26

Reply:
I've created the function `AssertFetchWithKeyMustExist` and used it, lmkwyt!